clients/unshaded/src/main/java/tachyon/client/file/options/SetAclOptions.java

```diff
@@ -0,0 +1,222 @@
+
+    /**
+     * Creates a new builder for {@link SetAclOptions}.
+     */
+    public Builder() {
+      this(ClientContext.getConf());
+    }
+
+    /**
+     * Creates a new builder for {@link SetAclOptions}.
+     *
+     * @param conf a Tachyon configuration
+     */
+    public Builder(TachyonConf conf) {
+      mOwner = null;
+      mGroup = null;
+      mPermission = Constants.INVALID_PERMISSION;
+      mRecursive = false;
+    }
+
+    /**
+     * Sets the owner of a path.
+     *
+     * @param owner to be set as the owner of a path
+     * @return the builder
+     */
+    public Builder setOwner(String owner) {
+      mOwner = owner;
+      return this;
+    }
+
+    /**
+     * Sets the group of a path.
+     *
+     * @param group to be set as the group of a path
+     * @return the builder
+     */
+    public Builder setGroup(String group) {
+      mGroup = group;
+      return this;
+    }
+
+    /**
+     * Sets the permission of a path.
+     *
+     * @param permission to be set as the permission of a path
+     * @return the builder
+     */
+    public Builder setPermission(short permission) {
+      mPermission = permission;
+      return this;
+    }
+
+    /**
+     * Sets the recursive flag.
+     *
+     * @param recursive whether to set acl recursively under a directory
+     * @return the builder
+     */
+    public Builder setRecursive(boolean recursive) {
+      mRecursive = recursive;
+      return this;
+    }
+
+    /**
+     * Builds a new instance of {@link SetAclOptions}.
+     *
+     * @return a {@link SetAclOptions} instance
+     * @throws IllegalArgumentException if the options are invalid
+     */
+    @Override
+    public SetAclOptions build() {
+      SetAclOptions options = new SetAclOptions(this);
+      if (options.isValid()) {
+        return options;
+      }
+      throw new IllegalArgumentException(
+          ExceptionMessage.INVALID_SET_ACL_OPTIONS.getMessage(mOwner, mGroup, mPermission));
+    }
+  }
+
+  private final String mOwner;
+  private final String mGroup;
+  private final short mPermission;
+  private final boolean mRecursive;
+
+  /**
+   * Constructs a new method option for setting the acl.
+   *
+   * @param options the options for setting the acl
+   */
+  public SetAclOptions(SetAclTOptions options) {
+    mOwner = options.isSetOwner() ? options.getOwner() : null;
+    mGroup = options.isSetGroup() ? options.getGroup() : null;
+    mPermission =
+        options.isSetPermission() ? (short) options.getPermission() : Constants.INVALID_PERMISSION;
+    mRecursive = options.isSetRecursive() ? options.isRecursive() : null;
+  }
+
+  private SetAclOptions(Builder builder) {
+    mOwner = builder.mOwner;
+    mGroup = builder.mGroup;
+    mPermission = builder.mPermission;
+    mRecursive = builder.mRecursive;
+  }
+
+  /**
+   * Checks whether the instance of {@link SetAclOptions} is valid, which means only one of three
+   * attributes (owner, group, permission) takes effect.
```

Comment:
Since generally we use `chown`, `chgrp`, `chmod` to set acl, each command's options will have only one effective attribute: owner, group, or mode.